File: backend/ASP.NET/EgyptHomes/DTOs/PropertySearchDTO.cs
using EgyptHomes.Models;
namespace EgyptHomes.DTOs;

public class PropertySearchDTO
{
    public string? City { get; set; }
    public string? Governorate { get; set; }
    public int? MinBathrooms { get; set; }
    public int? MinBedrooms { get; set; }
    public string? Street { get; set; }
    public decimal? MinPrice { get; set; }
    public decimal? MaxPrice { get; set; }
    public PropertyType? PropertyType { get; set; }
    public PropertyCategory? PropertyCategory { get; set; }
    public bool HasGarage { get; set; } = false;
    public bool HasSwimmingPool { get; set; } = false;
    public bool HasGarden { get; set; } = false;
    public bool HasAirConditioning { get; set; } = false;
    public int PageNumber { get; set; } = 1;
    public int PageSize { get; set; } = 10;
}

File: backend/JavaSpringBoot/egypthomes/src/main/java/com/egypthomes/dto/PropertySearchDTO.java
package com.egypthomes.dto;

import com.egypthomes.model.PropertyCategory;
import com.egypthomes.model.PropertyType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PropertySearchDTO {
    @Builder.Default
    private String city = null;
    @Builder.Default
    private String governorate = null;
    private Integer minBathrooms;
    private Integer minBedrooms;
    @Builder.Default
    private String street = null;
    private Double minPrice;
    private Double maxPrice;
    private PropertyType propertyType;
    private PropertyCategory propertyCategory;
    @Builder.Default
    private Boolean hasGarage = null;
    @Builder.Default
    private Boolean hasSwimmingPool = null;
    @Builder.Default
    private Boolean hasGarden = null;
    @Builder.Default
    private Boolean hasAirConditioning = null;
    @Builder.Default
    private Integer minArea = null;
}
